pilot/pkg/networking/core/v1alpha3/route/retry/retry_test.go

```diff
@@ -175,3 +178,38 @@ func TestMissingPerTryTimeoutShouldReturnNil(t *testing.T) {
 	g.Expect(policy).To(Not(BeNil()))
 	g.Expect(policy.PerTryTimeout).To(BeNil())
 }
+
+func TestRetryRemoteLocalities(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	// Create a route with a retry policy with RetryRemoteLocalities enabled.
+	route := networking.HTTPRoute{
+		Retries: &networking.HTTPRetry{
+			Attempts: 2,
+			RetryRemoteLocalities: &gogoTypes.BoolValue{
+				Value: true,
+			},
+		},
+	}
+
+	policy := retry.ConvertPolicy(route.Retries)
+	g.Expect(policy).To(Not(BeNil()))
+	g.Expect(policy.RetryPriority).To(Not(BeNil()))
+	g.Expect(policy.RetryPriority.Name).To(Equal("envoy.retry_priorities.previous_priorities"))
+	g.Expect(policy.RetryPriority.ConfigType).To(Not(BeNil()))
+
+	typedConfig, ok := policy.RetryPriority.ConfigType.(*envoyroute.RetryPolicy_RetryPriority_TypedConfig)
+	g.Expect(ok).To(Equal(true))
+
+	name := "type.googleapis.com/envoy.config.retry.previous_priorities.PreviousPrioritiesConfig"
+
+	g.Expect(typedConfig).To(Not(BeNil()))
+	g.Expect(typedConfig.TypedConfig).To(Not(BeNil()))
+	g.Expect(typedConfig.TypedConfig.TypeUrl).To(Equal(name))
+
+	data := &previouspriorities.PreviousPrioritiesConfig{}
+	err := proto.Unmarshal(typedConfig.TypedConfig.Value, data)
```

Comment:
Nit: this whole test would be a lot easier if you did something along the lines of `if !reflect.DeepEqual(ptypes.UnmarshalAny(policy), expected) { t.Fatal ... }`. Then we can just construct the expected output instead of having all of these different checks

Reply:
Good idea, done~